java/client/src/main/java/glide/api/models/commands/bitmap/BitFieldOptions.java

```diff
@@ -0,0 +1,187 @@
+        private final long increment;
+
+        /**
+         * Creates the INCRBY subcommand arguments.
+         *
+         * @return a String array with INCRBY arguments.
+         */
+        public String[] createArgs() {
+            return new String[] {
+                "INCRBY", encoding.getEncoding(), offset.getOffset(), Long.toString(increment)
+            };
+        }
+    }
+
+    /** OVERFLOW subcommand for {@link BitmapBaseCommands#bitfield(String, BitFieldSubCommands[])}. */
+    @RequiredArgsConstructor
+    public static class BitFieldOverflow implements BitFieldSubCommands {
+        private final BitOverflowControl overflowControl;
+
+        /**
+         * Creates the OVERFLOW subcommand arguments.
+         *
+         * @return a String array with OVERFLOW arguments.
+         */
+        public String[] createArgs() {
+            return new String[] {"OVERFLOW", overflowControl.toString()};
+        }
+
+        /** Supported overflow controls */
+        public enum BitOverflowControl {
+            WRAP,
+            SAT,
+            FAIL
+        }
+    }
+
+    private interface BitEncoding {
+        String getEncoding();
+    }
+
+    /** Specifies that the argument is a signed encoding */
+    public static final class SignedEncoding implements BitEncoding {
+        @Getter private final String encoding;
+
+        /**
+         * Constructor that prepends the number with "i" to specify that it is in signed encoding.
+         *
+         * @param encodingLength bit size of encoding.
+         */
+        public SignedEncoding(long encodingLength) {
+            encoding = "i".concat(Long.toString(encodingLength));
+        }
+    }
+
+    /** Specifies that the argument is a signed encoding */
+    public static final class UnsignedEncoding implements BitEncoding {
+        @Getter private final String encoding;
+
+        /**
+         * Constructor that prepends the number with "u" to specify that it is in unsigned encoding.
+         *
+         * @param encodingLength bit size of encoding.
+         */
+        public UnsignedEncoding(long encodingLength) {
+            encoding = "u".concat(Long.toString(encodingLength));
+        }
+    }
+
+    private interface BitOffset {
+        String getOffset();
+    }
+
+    /** Offset in the array of bits. */
+    public static final class Offset implements BitOffset {
+        @Getter private final String offset;
+
+        /**
+         * Constructor for Offset.
+         *
+         * @param offset element in the array of bits.
+         */
+        public Offset(long offset) {
+            this.offset = Long.toString(offset);
+        }
+    }
+
+    /** Offset in the array of bits multiplied by the encoding value. */
+    public static final class OffsetMultiplier implements BitOffset {
+        @Getter private final String offset;
+
+        /**
+         * Constructor that prepends the offset value with "#" to specify that it is a multiplier.
```

Comment:
The user don't have to know how it works under the hood.
```suggestion
         * Constructor for the offset multiplier.
```